corehq/apps/linked_domain/local_accessors.py

```diff
@@ -10,6 +10,40 @@
 from corehq.apps.users.models import SQLUserRole
 from corehq.apps.users.views.mobile import UserFieldsView
 from corehq.apps.integration.models import DialerSettings, GaenOtpServerSettings, HmacCalloutSettings
+from corehq.apps.reports.models import TableauServer, TableauVisualization
+
+
+def get_tableau_server_and_visualizations(domain):
+    server, created = TableauServer.objects.get_or_create(domain=domain)
+    visualizations = TableauVisualization.objects.all().filter(domain=domain).order_by('pk')
```

Comment:
This ordering is so that you can match up the upstream and downstream models, yeah?

Ordering seems fragile to rely on, especially when/if users delete visualizations. An alternative pattern is to add an `upstream_id` attribute to the model. This only gets populated for objects in a downstream domain, and it contains the pk of the matching upstream model.

`Keyword` and `UserRole` both do this - `UserRole` used to try to match up roles based on their name, but eventually we ran into issues with roles getting renamed/deleted and not matching, so we added upstream_id.

Reply:
Yeah, that was my thinking with the ordering. I see what you mean about it being fragile. I'll look at the implementation in `Keyword` and `UserRole` and try to do something similar.